fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -765,6 +739,201 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
+                            Set<String> queryIds = resourceTypeMap.get(includeParm.getJoinResourceType());
+
+                            // Build and run the query
+                            List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResources =
+                                    this.runIncludeQuery(resourceType, searchContext, queryBuilder, includeParm,
+                                        SearchConstants.INCLUDE, queryIds, queryResultMap, resourceDao, i+1, allResourceIds);
+
+                            // Add new ids to de-dup list
+                            allResourceIds.addAll(includeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                            // Add resources to list
+                            allIncludeResources.addAll(includeResources);
+
+                            // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                            if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
+                                return allIncludeResources;
+                            }
+                        }
+                    }
+                }
+                for (InclusionParameter revincludeParm : searchContext.getRevIncludeParameters()) {
+                    if (revincludeParm.isIterate() && resourceTypeMap.keySet().contains(revincludeParm.getSearchParameterTargetType())) {
+                        // For iteration 0, we only process if target type is same as join type
+                        if (i > 0 || revincludeParm.getJoinResourceType().equals(revincludeParm.getSearchParameterTargetType())) {
+                            // Get ids to query against
+                            Set<String> queryIds = resourceTypeMap.get(revincludeParm.getSearchParameterTargetType());
+
+                            // Build and run the query
+                            List<com.ibm.fhir.persistence.jdbc.dto.Resource> revincludeResources =
+                                    this.runIncludeQuery(resourceType, searchContext, queryBuilder, revincludeParm,
+                                        SearchConstants.REVINCLUDE, queryIds, queryResultMap, resourceDao, i+1, allResourceIds);
+
+                            // Add new ids to de-dup list
+                            allResourceIds.addAll(revincludeResources.stream().map(r -> r.getId()).collect(Collectors.toSet()));
+
+                            // Add resources to list
+                            allIncludeResources.addAll(revincludeResources);
+
+                            // Check if max size exceeded. If so, return results and let rest helper throw exception.
+                            if (allIncludeResources.size() > SearchConstants.MAX_PAGE_SIZE) {
+                                return allIncludeResources;
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
Could it also break out of the loop if no new resources were added during the current iteration?

Reply:
This is basically handled by the check on line 829. If no resources got added by a previous iteration, the resourceTypeMap will be null.

Reply:
Ah, ok. I wasn't able to piece that together at first glance. Thanks!